### PR TITLE
Development

### PR DESCRIPTION
**Pull request by RogueClaris**
Changes:

1. Revert my reckless usage of `tfEvents.end()-1` to `tfEvents.begin()` to properly re-allow Time Freeze Counter. My mistake, and I apologize.
2. Attempt at enabling card action animations while time frozen by disabling Freeze/Stun status effects, storing their timers, and re-applying them after time freeze is over. This works, but may not be what was asked for, so I am submitting it for review.
3. Game-accurate Time Freeze Counter mechanics; it's not if you're actionable, strictly speaking, but if you're not in a card lockout. You can actually TFC while in another animation, as long as you're not dead or in a card lockout of some kind.
4. Poison Tiles were draining Health every frame. Accuracy dictates it be every 7 frames. I am not sure if this was for debugging purposes, but I have set it back to every 7 frames.

### CODE REVIEW

**Comment by TheMaverickProgrammer** [comment]
I never had the correct poison timers so this is great! I'll take a look at the rest.

File: BattleNetwork/bnEntity.cpp [hunk 1]
```diff
@@ -1071,6 +1071,36 @@ void Entity::AdoptNextTile()
 
 void Entity::ToggleTimeFreeze(bool state)
 {
+  if (state) {
+    if (currentStatus.statusName != "None") {
```
Comment (TheMaverickProgrammer) [inline BattleNetwork/bnEntity.cpp:1075]: avoid using strings. not only are they large in size (and stored on the heap), but they are human-error prone. For instance if I type "none" on accident or say "NONe" while in a coding frenzy, we would lose our minds debugging such a simple mistake.

We already have bit flags used for HitProps. We can simply have an internal `Hit::Flag` member varible and check to see if the Freeze or Stun bits are set.

File: BattleNetwork/bnEntity.h [hunk 2]
```diff
@@ -95,6 +95,11 @@ struct EntityComparitor {
   bool operator()(Entity* f, Entity* s) const;
 };
 
+struct AppliedStatus {
+  std::string statusName;
```
Comment (TheMaverickProgrammer) [inline BattleNetwork/bnEntity.h:99]: change to use the bit flag

File: BattleNetwork/bnEntity.h [hunk 3]
```diff
@@ -132,6 +137,8 @@ class Entity :
   EventBus::Channel channel; /*!< Our event bus channel to emit events */
   MoveEvent currMoveEvent{};
   VirtualInputState inputState;
+  AppliedStatus previousStatus{ "None", frames(0) }; /*!< The previous status, Stun or Freeze, applied when time freeze started.*/
```
Comment (TheMaverickProgrammer) [inline BattleNetwork/bnEntity.h:140]: change these 2 to use the bit flag e.g. `{ Hit::None, frames(0) };`

**Review by TheMaverickProgrammer — CHANGES_REQUESTED** [review]
see comments

File: BattleNetwork/bnEntity.h (hunk 3 above)
Comment (RogueClaris) [reply BattleNetwork/bnEntity.h]: The requested changes have been made.

**Review by RogueClaris — COMMENTED** [review]
GitHub wants me to review these changes, even though they're outdated.

**Review by TheMaverickProgrammer — CHANGES_REQUESTED** [review]

File: BattleNetwork/bnStatusDirectior.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        return previousStatus; //Everything works, but it says Do Not Slice - seems to be a warning about data management. I can't seem to crack it.
+    }
+    return currentStatus;
+};
+
+
+StatusBehaviorDirector::~StatusBehaviorDirector()
+{
+    //Reset both statuses to default when deleting.
+    previousStatus = AppliedStatus{ Hit::none, frames(0) };
```
Comment (TheMaverickProgrammer) [inline BattleNetwork/bnStatusDirectior.cpp:27]: not needed

File: BattleNetwork/bnStatusDirectior.h [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+#include "bnHitProperties.h"
+#include "bnFrameTimeUtils.h"
+
+struct AppliedStatus {
+    Hit::Flags statusFlag;
+    frame_time_t remainingTime;
+};
+
+class StatusBehaviorDirector :
+    public AppliedStatus
```
Comment (TheMaverickProgrammer) [inline BattleNetwork/bnStatusDirectior.h:12]: no reason to inherit anything

File: BattleNetwork/bnStatusDirectior.h [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+{
+public:
+    StatusBehaviorDirector();
+    virtual ~StatusBehaviorDirector();
+    void SetNextStatus(Hit::Flags statusFlag, frame_time_t maxCooldown, bool deffer);
+    AppliedStatus GetStatus(bool isPrevious);
+    AppliedStatus previousStatus{ Hit::none, frames(0) };
```
Comment (TheMaverickProgrammer) [inline BattleNetwork/bnStatusDirectior.h:19]: make these private if you have a `GetStatus` utility

**Review by TheMaverickProgrammer — CHANGES_REQUESTED** [review]

**Review by TheMaverickProgrammer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**